src/modules/Builder/features/ActivityFlowBuilder/ActivityFlowBuilder.tsx

```diff
@@ -52,7 +52,12 @@ export const ActivityFlowBuilder = () => {
     name: `${activityFlowName}.items`,
     keyName: REACT_HOOK_FORM_KEY_NAME,
   });
-  const activities: AppletFormValues['activities'] = watch('activities');
+  const formActivities: AppletFormValues['activities'] = watch('activities');
```

Comment:
here and in other places, please consider to use [useWatch](https://react-hook-form.com/docs/usewatch) :
```suggestion
  const formActivities: AppletFormValues['activities'] = useWatch('activities');
```
watch - This API will trigger re-render at the root of your app or form, consider using a callback or the useWatch api if you are experiencing performance issues.